### ISSUE TITLE
How to use 'instead of http:// use yourscheme:// as a link.'

### ISSUE THREAD

**Issue by pendave** [SanderRonde/CustomRightClickMenu#9]
Hi, I generate a URI pair with clicking the generate button and get a text file downloaded.

Then I change it to .reg file and run it.

```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\BandicamPortable]
@="URL:BandicamPortable Protocol"
"URL Protocol"=""

[HKEY_CLASSES_ROOT\BandicamPortable\shell]

[HKEY_CLASSES_ROOT\BandicamPortable\shell\open]

[HKEY_CLASSES_ROOT\BandicamPortable\shell\open\command]
@="\"D:\\greensoft\\BandicamPortable\\BandicamPortable.exe\""
```

What should I do now?

To add a link to the CustomRightClickMenu plugin?

![qq 20170509233112](https://cloud.githubusercontent.com/assets/1331881/25858981/c4dd2920-350f-11e7-8fd8-0543c34e50b2.png)

But the Chrome just download this .exe file when I right click it, instead of running the program.

**Comment by SanderRonde** [comment]
Instead of putting the file path in there, you need to put (in your case) "BandicamPortable Protocol://" into the link box (so not the link to the file path, but to the URI scheme). However, "BandicamPortable Protocol" won't work because it contains a space (which should have been marked as an invalid scheme). So first of all you need to generate a new scheme that does not contain spaces, like myscheme (the standard name). Then in the "link(s)" box you need to put "myscheme://", and then it should launch your program.

**Comment by pendave** [comment]
How to generate a new scheme like "myscheme://" ?

Any tutorial for that?

**Comment by SanderRonde** [comment]
The same way you generated the other one, by clicking the "generate" button after entering what you want into the inputs. Except this time entering "myscheme" (be sure to not include the :// in the name, only do that while using it as a link) instead of "BandicamPortable Protocol".

**Comment by pendave** [comment]
Great! It works!

![6 lq4msnlase hzhfi3_98](https://cloud.githubusercontent.com/assets/1331881/25881158/01464352-356e-11e7-8bd4-a36917231e11.png)

Then I need to remove the former error .reg info.

**Comment by pendave** [comment]
I think you should record some sample tutorials for using the plugin you created, and upload them to Youtube so that guys will handle the tips well.

**Comment by SanderRonde** [comment]
You don't really have to remove it, it won't get accidentally activated anyway and can't do any harm except launching your bandicam if someone decides to target you specifically. 

If you really want to remove it though, you can go to "HKEY_CLASSES_ROOT\BandicamPortable\shell\open\command" in your regedit and remove the key

Regarding creating tutorials, i might do that eventually but i'm just a bit busy at the moment so it might take a while.

**Comment by pendave** [comment]
BTW, can I use the same "myscheme://" to open other .exe applications?

like D:\greensoft\PDFBinderPortable\PDFBinder.exe

**Comment by SanderRonde** [comment]
You'd have to choose a new name and generate a new regex file, otherwise it won't know which one to open, so for example use PDFBinder as a name instead of myscheme.

**Comment by pendave** [comment]
I see, after I try to add another exe behind it opend the default bandicam exe.

![aridvl y7uc u 9wl 7r3](https://cloud.githubusercontent.com/assets/1331881/25881567/49bd803a-3570-11e7-820b-79d135b3b964.png)

**Comment by SanderRonde** [comment]
Thats because the url you're linking to in the address bar still starts with myscheme:// and not the new name (don't know what it is as I can't read chinese). 
By the way what comes after the "://" does not matter, it opens a program/file based on what comes before it, so simply linking do myscheme:// does the same as linking to myscheme://D://Some/U/R/L.exe.

You are registering a url scheme (like http:// or https://) to open a certain programs, linking only those things, not what comes after it.

**Comment by pendave** [comment]
Could this kind of running exe or bat or cmd get parameters passed?

If I want to right click on a Youtube page to pass the url to python.exe?

**Comment by SanderRonde** [comment]
Yes, if you read this page you can learn more about it https://msdn.microsoft.com/en-us/library/aa767914%28VS.85%29.aspx?f=255&MSPPError=-2147217396. However the extension can not generate this one for you, you have to edit the .reg file (with notepad or something) and behind the .exe\" place this:
"%1"

So the last line of your earlier examples becomes:
@="\"D:\\greensoft\\BandicamPortable\\BandicamPortable.exe\" "%1""

After that anything after the :// in the link is passed, so myscheme://abc launches:
BandicamPortable.exe abc

However you can not as you said click on a youtube page and pass that specific URL. You can only pass predefined data (so like the links). If you would want to do that you would have to open the URI with javascript which chrome forbids. 

Basically if you want to learn more or don't understand, the page i linked explains this much better than i can.
